### PR TITLE
Enable bz2 compression

### PR DESCRIPTION
**Pull request by xhochy**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -27,6 +27,7 @@ requirements:
   host:
     - boost-cpp
     - brotli
+    - bzip2
```
Comment (pitrou) [inline recipe/meta.yaml]: Does it need to be added to the "run" dependencies as well? I forget whether we link statically by default...

Comment (xhochy) [reply recipe/meta.yaml]: No, the `bzip2` package has a `run_exports` section that defines what goes into `run` of the package that uses it in the `host` section. To check this, see:

```
% conda render recipe | grep 'run:' -A 20 | grep bzip2
        - bzip2 >=1.0.8,<2.0a0
```

Comment (pitrou) [reply recipe/meta.yaml]: Ah, is this a new conda thing?
But if we were to link statically, it would still appear as a runtime dependency?

Comment (xhochy) [reply recipe/meta.yaml]: Yes, I guess it came with 4.6. 

If we link statically, it will still appear but we should link dynamically on conda-forge.

Comment (xhochy) [reply recipe/meta.yaml]: Build output shows we're linking dynamically:

```
   INFO (arrow-cpp,lib/libarrow.so.14.1.0): Needed DSO lib/libbz2.so.1.0 found in conda-forge::bzip2-1.0.8-h516909a_1
```

**Comment by xhochy** [comment]
@pitrou Ok to merge this?

**Pull request merged**